troubleshoot-tidb-oom.md

```diff
@@ -0,0 +1,198 @@
+
+#### Executing SQL statements consumes too much memory
+
+You can take the following measures to reduce the memory usage of SQL statements, depending on the different causes of OOM issues.
+
+- If the execution plan of SQL is not optimal, for example, due to lack of proper indexes, outdated statistics, or optimizer bugs, a wrong execution plan of SQL might be selected. A huge intermediate result set will then be accumulated in the memory. In this case, consider the following measures.
+    - Add appropriate indexes
+    - Use the [disk spill](/configure-memory-usage.md#disk-spill) feature for execution operators
+    - Adjust the JOIN order between tables
+    - Use hints to optimize SQL statements
+
+- Some operators and functions do not support pushing down to the storage level, resulting in a huge accumulation of intermediate result sets. In this case, you need to refine the business SQL statements or use hints to tune, and use the functions or operators that support pushing down.
+
+- The execution plan contains the operator HashAgg. HashAgg is executed concurrently by multiple threads, which is faster but consumes more memory. Instead, you can use `STREAM_AGG()`.
+
+- Reduce the number of regions to be read simultaneously or reduce the concurrency of operators to avoid memory problems caused by high concurrency. The corresponding system variables include:
+    - [`tidb_distsql_scan_concurrency`](/system-variables.md#tidb_distsql_scan_concurrency)
+    - [`tidb_index_serial_scan_concurrency`](/system-variables.md#tidb_index_serial_scan_concurrency)
+    - [`tidb_executor_concurrency`](/system-variables.md#tidb_executor_concurrency-span-classversion-marknew-in-v50span)
+
+- The concurrency of sessions is too high near the time point when the problem occurs. In this case, consider scaling out the TiDB cluster by adding more TiDB nodes.
+
+#### Large transactions or large writes consume too much memory
+
+You need to plan in advance for memory capacity. TiDB can consume up to two to three times more memory than the size of a committed transaction when processing a transaction.
+
+You can split a single large transaction to make it smaller in size.
+
+#### The process of collecting and loading statistical information consumes too much memory
+
+A TiDB node needs to load statistics into memory after it starts. TiDB consumes memory when collecting statistical information. You can control memory usage by the following ways:
+
+- Specify a sampling rate, only collect statistics for specific columns, and reduce analyze concurrency.
+- Since TiDB v6.1.0, you can use the system variable [`enable_tidb_stats_cache_mem_quota`](/tidb-configuration-file.md#enable-stats-cache-mem-quota-span-classversion-marknew-in-v610span) to control the memory usage for statistical information.
```

Comment:
```suggestion
- Since TiDB v6.1.0, you can use the system variable [`tidb_stats_cache_mem_quota`](/tidb-configuration-file.md#tidb-stats-cache-mem-quota-span-classversion-marknew-in-v610span) to control the memory usage for statistical information.
```